Fix: references to the 'Get Started' UI label

max-cx opened the pull request:
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Fixed references to the 'Get Started' UI label (issue 2999) and rewrote several files in the docs.

### What issues does this PR fix or reference?
2999

### Specify the version of the product this PR applies to.
Che 7.30.x

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been configured on the docs side):
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

max-cx committed:
salvaging earlier versions of some of modified files after the vale > wipeout

max-cx committed:
removing the files deleted during editing as part of the salvage operation

max-cx committed:
pre-Vale, salvaged and rewritten

max-cx committed:
fixed editing-the-commands-after-importing-a-project

max-cx committed:
fixing a missed {prod-short}

max-cx committed:
merging upstream changes

max-cx committed:
added text to one link in proc_adding-language-support-plug-in-to-the-che-workspace.adoc

max-cx committed:
renamed  as

max-cx committed:
minor word fixes

max-cx committed:
replaced one word in a section title

max-cx committed:
minor text fixes to some files

max-cx commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -54,25 +66,37 @@ projects:
-* Select your workspace by selecting its name. This will link you to the workspace's *Overview* tab.
-* Open *Devfile* tab where you can enter your own YAML configuration.
-* Add the project.
+You can import your project into a created workspace by editing its devfile.
```
Perhaps I could replace `its` with either `the project's` or `the workspace's`.

Eva-Lotte replied:
I think all three options work. Go for the one you like :)

themr0c committed:
Merge branch 'master' into 2999max-cx

themr0c commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -107,35 +148,54 @@ image::workspaces/configure-command.png[Configure tasks]
 
-In addition to the approaches above, you can also start a workspace, open a *Terminal*, and type `git clone` to pull code.
+.Prerequisites
+* A started workspace
```
```suggestion
* A running workspace
```

max-cx committed:
Update assembly_creating-a-workspace-from-included-options.adoc

troubleshooting an issue with *Validate links*

max-cx committed:
Update modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc

Thanks.

Co-authored-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

max-cx commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -54,25 +66,37 @@ projects:
-====
-To add a project into the workspace, add or edit the following section:
+.Prerequisites
+* A started workspace
```
```suggestion
* A running workspace
```

max-cx committed:
Update modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc

MichalMaler committed:
Merge branch 'master' into 2999max-cx

Eva-Lotte commented on modules/end-user-guide/partials/assembly_authoring-devfiles-version-2.adoc:
```diff
@@ -1,21 +1,26 @@
-.Prerequisites:
-* An instance of {prod-short} with the DevWorkspace Operator enabled, see xref:installation-guide:enabling-dev-workspace-engine.adoc[]
+.Prerequisites
+* An instance of {prod-short} with the xref:installation-guide:enabling-dev-workspace-engine.adoc[DevWorkspace Operator enabled]
```
a full-stop is missing.

Eva-Lotte replied:
I would move "enabled" out of the square brackets.

max-cx replied:
> a full-stop is missing.

I left it out on purpose, as it is just a list item that is not a sentence. Is that a problem with the IBM Style Guide?

max-cx replied:
> I would move "enabled" out of the square brackets.

Thanks, done that now.

Eva-Lotte commented on modules/end-user-guide/partials/assembly_authoring-devfiles-version-2.adoc:
```diff
@@ -1,21 +1,26 @@
 :context: authoring-devfiles-version-2
 
-This section explains the concept of the devfile 2.0 specification and how to configure a {prod-short} workspace by using a devfile 2.0. 
+Your devfile for configuring a workspace must meet the devfile 2.1.0 specification.
```
This sentence doesn't make it entirely clear what this section is about or what the user should do. After reading it my first thought was "so?".
Maybe you could add "To update your devfile to meet the specifications/to version 2.1.0 etc. whatever makes sense, complete the following steps"

max-cx replied:
Thanks, I tweaked the text a bit in both the intro sentence and in the procedure sentence.

Eva-Lotte commented on modules/end-user-guide/partials/assembly_authoring-devfiles-version-2.adoc:
```diff
@@ -1,21 +1,26 @@
 //TODO - Figure out whether we want to duplicate, single source or just reference devfile docs here
-* To create devfiles with the 2.0 specification, see link:https://docs.devfile.io/devfile/2.0.0/user-guide/authoring-stacks.html[]
+
+Create a devfile 2.1.0 with the help of the link:https://docs.devfile.io/devfile/2.0.0/user-guide/authoring-stacks.html[Devfile User Guide].
```
I'd add a bullet point.

Eva-Lotte replied:
May I ask why the procedure is a link to somewhere else? I'm not so sure about your workflows and such, but shouldn't we document the steps ourselves and move the devfiles user guide to additional resources for deeper understanding?

max-cx replied:
> I'd add a bullet point.

Thanks, done that.

max-cx replied:
> May I ask why the procedure is a link to somewhere else? I'm not so sure about your workflows and such, but shouldn't we document the steps ourselves and move the devfiles user guide to additional resources for deeper understanding?

No idea. i guess they ended up with two devfile format versions to mention, so this resulted in two assembly files, one for each. I pretty much just reworded the text to make it a bit more readable.

Eva-Lotte replied:
I'd inquire with the SMEs about it. This is really not the way it should be.

max-cx replied:
> I'd inquire with the SMEs about it. This is really not the way it should be.

Yep, I plan to.

Eva-Lotte commented on modules/end-user-guide/partials/assembly_authoring-devfiles-version-2.adoc:
```diff
@@ -1,21 +1,26 @@
 .Additional resources
 
-* link:https://redhat-developer.github.io/devfile/devfile[Devfile 2.0 specification]
+* link:https://redhat-developer.github.io/devfile/devfile[Introduction to Devfile]
```
We usually add some text to the additional resources links, like "For more information on xy, see [], or "to download [] to do xy".
In that case, the link ends with a full-stop.

max-cx replied:
I like to do exactly that for inline references, but this instance makes me wonder if that's worth a sentence if it's already listed under `.Additional resources` and the heading on the link is self-explanatory.

Eva-Lotte replied:
Yes, we do that in additional resources, it's custom as I commented before.

max-cx replied:
Alright, thanks, I added a sentence in there.

Eva-Lotte commented on modules/end-user-guide/partials/assembly_authoring-devfiles-version-2.adoc:
```diff
@@ -1,21 +1,26 @@
 
 [id="authoring-devfiles-version-2_{context}"]
-= Authoring devfiles version 2
+= Authoring a devfile 2.1.0
```
if 2.1.0 is a version I'd add "version" to the title for clarity.

max-cx replied:
I was thinking of adding "version" there, but was hesitant to add "version" there for these reasons:
1. The "version" in this text refers to the devfile's format or specification; the devfile itself has no "version 1" or "version 2".
2. Having a title on "authoring a devfile version" seems a bit distracting to me, as it's about authoring a devfile, not about authoring a version.
3. The IMB Style Guide on p. 185 seems to suggest not using "version" when referring to a product name and alternatively using only "version" on its own as a generic reference without the product name.

Eva-Lotte commented on modules/end-user-guide/partials/assembly_creating-a-workspace-from-included-options.adoc:
```diff
@@ -0,0 +1,20 @@
+
+:context: creating-a-workspace-from-included-options
+
+{prod-short} includes optional workspace samples and devfile templates to simplify workspace creation.
```
"to simplify your workspace creation/the creation of a workspace". According to IBM Style Guide p. 45 we are supposed to not omit articles.

max-cx replied:
I left it without an article here because I used "workspace" as a noun in place of an adjective, and "creation" can be both countable and uncountable, with my tendency to use it as uncountable when describing the process of creating something. I've added an `a` to make it refer to a single workspace. Other that this, I'll probably have to discuss opening a separate issue for omitted articles because there are many missing articles in headings given the IMB Style Guide preference that you referenced.

Eva-Lotte commented on modules/end-user-guide/partials/assembly_creating-a-workspace-from-included-options.adoc:
```diff
@@ -0,0 +1,20 @@
+
+{prod-short} includes optional workspace samples and devfile templates to simplify workspace creation.
+
+. xref:creating-a-workspace-from-workspace-sample_{context}[Creating a workspace from workspace sample]
```
In the docs I'm working on the beginning of the assembly should have an introduction and optionally a prerequisites section (https://redhat-documentation.github.io/modular-docs/#forming-assemblies). I'm not sure what the sentence and the two xrefs should mean. I think that could be because of the differences in our docs, but I'm not sure. When you read this comment you'll know :)

max-cx replied:
Actually, this is just my edited version of another file that I deleted as part of this PR:
https://github.com/eclipse/che-docs/blob/73e8c09106f6e83893ff5fa5b806522963e07e0c/modules/end-user-guide/partials/assembly_creating-a-workspace-from-code-sample.adoc

Eva-Lotte commented on modules/end-user-guide/partials/assembly_creating-a-workspace-from-included-options.adoc:
```diff
@@ -0,0 +1,20 @@
+
+{prod-short} includes optional workspace samples and devfile templates to simplify workspace creation.
+
+. xref:creating-a-workspace-from-workspace-sample_{context}[Creating a workspace from workspace sample]
```
from *a* workspace sample

max-cx replied:
Yep, as I already mentioned, I'll have to discuss opening a separate issue for omitted articles because there are many missing articles in headings and file names. I've spent quite a lot of time on redoing some headings and file names; on the bright side, that was getting the experience I will need.

max-cx replied:
Eva, do you know if articles are included or skipped in the file names of pages, assemblies, and partials?

Eva-Lotte commented on modules/end-user-guide/partials/assembly_workspaces-overview.adoc:
```diff
@@ -7,7 +7,7 @@
 :context: developer-workspaces
 
-{prod} provides developer workspaces with everything needed to a code, build, test, run, and debug applications. To allow that, the developer workspaces provide four main components:
+{prod} provides developer workspaces with everything you need to code, build, test, run, and debug applications. To allow that, the developer workspaces provide four main components:
```
Suggestion:
"{prod} provides developer workspaces with everything you need to code, build, test, run, and debug applications.
Developer workspaces consist of four main components:"
Otherwise you'll have the prod that provides workspaces and the workspaces that provide components and that's weirdly nested. Also, "to allow that" is vague.

max-cx replied:
Thank you, Eva. I've just rewritten this. Could you please have a second look at it?
https://github.com/eclipse/che-docs/blob/c133efeb7902f83df21cb6b681ed4d8a6c8b2f5a/modules/end-user-guide/partials/assembly_workspaces-overview.adoc

max-cx committed:
fixing up the related page files

max-cx committed:
fixing up related page files

max-cx committed:
fixing up related page files

max-cx committed:
Merge remote-tracking branch 'upstream/master' into 2999max-cx

max-cx committed:
edited proc_creating-a-workspace-from-workspace-sample.adoc and removed link to deleted image

max-cx committed:
minor fix in another file - same as previous commit

Eva-Lotte commented on modules/end-user-guide/partials/proc_adding-language-support-plug-in-to-the-che-workspace.adoc:
```diff
@@ -8,6 +8,7 @@
 This procedure describes adding a tool to an existing workspace by enabling a dedicated plug-in from the Dashboard.
```
The Language Support for Java Plug-ins with an s. Also, is this a chapter in the guide? Or what is it? It should be mentioned. For example "The Language Support for Java Plug-ins" chapter. Also "" or italics would make sense in that case.

max-cx replied:
Thank you, Eva, but I'll just add the missing `s` here and move on - I've spent too much time fixing the docs outside of the issue scope already. Unfortunately, I can't fix it all at once.

Eva-Lotte commented on modules/end-user-guide/partials/proc_adding-language-support-plug-in-to-the-che-workspace.adoc:
```diff
@@ -18,8 +19,9 @@ This procedure uses the Language Support for Java plug-in as an example.
 * A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
```
Are those prerequisites? Maybe the problem is the rich diff again, but those bullet points seem to have no context.

max-cx replied:
Yep, this makes more sense when I open the file in a separate tab.
I improved it a bit just now anyway.

Eva-Lotte commented on modules/end-user-guide/partials/proc_adding-language-support-plug-in-to-the-che-workspace.adoc:
```diff
@@ -18,8 +19,9 @@ This procedure uses the Language Support for Java plug-in as an example.
 * A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
 
 * An existing workspace defined in this instance of {prod}; see:
```
"For more information, see" is more clear. Because you should tell the user what the condition is in which they should follow the link.

max-cx replied:
Done.

Eva-Lotte commented on modules/end-user-guide/partials/proc_adding-language-support-plug-in-to-the-che-workspace.adoc:
```diff
@@ -18,8 +19,9 @@ This procedure uses the Language Support for Java plug-in as an example.
+** xref:creating-a-workspace-from-included-options.adoc#creating-a-workspace-from-workspace-sample_{context}[Creating a workspace from workspace sample]
 
  * The workspace must be in a **stopped** state. To stop a workspace:
 
```
I'd remove "(the workspace's icon on the list will turn grey)" because it's not necessary and fluff, also it should be "To add a plug-in from the Plug-in registry", a not the. I realise these lines have nothing to do with your PR, but since they're in the guide and I came across them I thought I'd comment.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -3,49 +3,61 @@
 .Prerequisites
 
-* A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
+* A running instance of {prod-short}
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -3,49 +3,61 @@
+
+With {prod-short}, you can import and work on your project's current codebase in a workspace.
+
+You can import your project's current codebase into a workspace with any of the following options:
```
maybe "using" instead of "with"?

max-cx replied:
Done, thanks.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -3,49 +3,61 @@
+* xref:creating-a-custom-workspace-from-the-dashboard_{context}[Importing a project when creating a workspace]
+* xref:importing-from-the-dashboard-into-an-existing-workspace_{context}[Importing a project into a created workspace]
+* xref:importing-to-a-running-workspace-using-the-git-clone-command_{context}[Importing a project with *Git: Clone* in the GUI]
+* xref:importing-to-a-running-workspace-with-git-clone-in-a-terminal_{context}[Importing a project with `git clone` in a terminal]
```
I'd use "the" terminal to make it consistent with GUI, also I think it's clear which terminal is meant.

max-cx replied:
I had to decide between 'a' and 'the' and in the end settled on 'a' due to the fact that in the GUI the user can open multiple terminal windows, terminals, and use any of them to perform this procedure.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -3,49 +3,61 @@
+To install {prod-short}, see the xref:installation-guide:installing-che.adoc[Installation Guide].
 
-* An existing workspace with plug-ins related to your development environment defined on this instance of {prod} xref:configuring-a-workspace-with-dashboard.adoc[].
+* A configured workspace with plug-ins related to your development environment defined on this instance of {prod} xref:configuring-a-workspace-with-dashboard.adoc[Configuring a {prod-short} {prod-ver} workspace]
```
{prod}. For more information, see ...

max-cx replied:
Thanks for spotting that, fixed!

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -3,49 +3,61 @@
+. Enter a *Workspace Name* for the workspace that you are creating.
++
+[NOTE]
+.New workspace name
```
I've never seen a note with a title, and the note would definitely work without it. But maybe you have different guidelines.

max-cx replied:
Thanks, good to know. I have removed this note title in the few files where I found it.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -54,25 +66,37 @@ projects:
-====
-To add a project into the workspace, add or edit the following section:
+.Prerequisites
+* A running workspace
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -54,25 +66,37 @@ projects:
+
+. Go to the xref:navigating-che-using-the-dashboard.adoc[Dashboard] of {prod-short}.
+
+. Under *RECENT WORKSPACES*, hover over the name of the workspace to which to import the project.
```
maybe "to which you wish to import the project"? To be more user-centric

max-cx replied:
Thank you :). I rewrote it as "to which you will import the project".

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -81,19 +105,36 @@ projects:
+
+.Prerequisites
+
+* A workspace with a project in it
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -107,35 +148,54 @@ image::workspaces/configure-command.png[Configure tasks]
 image::workspaces/git-clone-command.png[Invoke git clone command]
 
-. Enter the path to the project that is about to be cloned.
+. Enter the path to the project to be cloned.
```
"you want to clone", would be active voice and more user centric.

max-cx replied:
Thank you, done just that.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -107,35 +148,54 @@ image::workspaces/configure-command.png[Configure tasks]
+You can import your project into a running workspace with the *Git: Clone* command in the graphic user interface of {prod-short}.
+ 
+.Prerequisites
+* A started workspace
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -107,35 +148,54 @@ image::workspaces/configure-command.png[Configure tasks]
 
-In addition to the approaches above, you can also start a workspace, open a *Terminal*, and type `git clone` to pull code.
+.Prerequisites
+* A running workspace
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -107,35 +148,54 @@ image::workspaces/configure-command.png[Configure tasks]
 Importing or deleting workspace projects in the terminal does not update the workspace configuration, and the IDE does not reflect the changes in the *Devfile* tab in the dashboard.
 
-Similarly, when you add a project using the *Dashboard*, then delete it with `rm -fr myproject`, it may still appear in the *Devfile* tab.
+Similarly, when you add a project using the *Dashboard* and then delete it with `rm -fr myproject`, it may still appear in the *Devfile* tab.
```
"the project may still appear" would be very clear.

max-cx replied:
Yep, rewrote that sentence.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-template-devfile.adoc:
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+
+[id="creating-a-workspace-from-template-devfile_{context}"]
+= Creating a workspace from template devfile
```
from a devfile template or from a template devfile, but definitely an "a" is missing.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-template-devfile.adoc:
```diff
@@ -0,0 +1,32 @@
+
+.Prerequisites
+
+* A running instance of {prod-short}
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-template-devfile.adoc:
```diff
@@ -0,0 +1,32 @@
+
+. In the left navigation panel, click btn:[Create Workspace] to go to the *Custom Workspace* tab.
+
+. Enter a *Workspace Name* for the workspace that you are creating.
```
can't you just say "enter a workspace name" without the markup?

max-cx replied:
Yep, I should have rewritten it earlier.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-template-devfile.adoc:
```diff
@@ -0,0 +1,32 @@
+. Enter a *Workspace Name* for the workspace that you are creating.
++
+[NOTE]
+.New workspace name
```
again, notes with titles are something I didn't see before.

max-cx replied:
Neither have I (new to AsciiDoc). I removed the title thanks to your feedback!

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-workspace-sample.adoc:
```diff
@@ -0,0 +1,30 @@
+// creating-a-workspace-from-included-options
+
+[id="creating-a-workspace-from-workspace-sample_{context}"]
+= Creating a workspace from workspace sample
```
from "a" workspace sample. We are not to omit articles to shorten things. *shrug*

max-cx replied:
Yep, I just fixed this now, but the docs still have more of omitted articles in the headings.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-workspace-sample.adoc:
```diff
@@ -0,0 +1,30 @@
+
+.Prerequisites
+
+* A running instance of {prod-short}
```
.

max-cx replied:
Period omitted deliberately. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-workspace-sample.adoc:
```diff
@@ -0,0 +1,30 @@
+
+. Go to the xref:navigating-che-using-the-dashboard.adoc[Dashboard] of {prod-short}.
+
+. In the left navigation panel, click btn:[Create Workspace] to go to the *Get Started* tab with workplace samples.
```
containing workplace samples?

max-cx replied:
I initially wrote it as "... tab that contains workspace samples", so rewriting it as such now.

Eva-Lotte commented on modules/end-user-guide/partials/proc_creating-a-workspace-from-workspace-sample.adoc:
```diff
@@ -0,0 +1,30 @@
+[NOTE]
+.New workspace name
+====
+{prod-short} generates a unique name for a new workspace from the devfile's `metadata.generateName` property followed by four random characters. You can edit the name of the workspace when it is created.
```
Out of curiosity, why is the note different three times?

max-cx replied:
In some cases, a value for this variable is automatically generated by default without the user's input; in other cases, the user is offered to enter it manually by default. So these notes discuss the same thing but in different contexts.

max-cx committed:
added an indefinite article

max-cx committed:
minor fixes in the text

max-cx committed:
added a block comment for future reference

max-cx committed:
updated the block comment

max-cx committed:
minor text changes to improve readability

max-cx committed:
Added a sentence for the additional resource

max-cx committed:
added a 'the'

max-cx committed:
Update assembly_creating-a-workspace-from-included-options.adoc

max-cx committed:
rewrote the intro par

max-cx committed:
fixing a missing plural

max-cx committed:
minor edits in .Prerequisites

max-cx committed:
minor fixes

max-cx committed:
minor edits

max-cx committed:
minor edit

MichalMaler committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
rewrote the procedure

also, replaced 'existing' with 'created', because there is a procedure to create a workspace

max-cx committed:
rewrote the procedure even more

max-cx committed:
minor edits

max-cx committed:
minor edits

max-cx committed:
rewrote the last step in the procedure

max-cx committed:
corrected "2.1.0" > "2.0.0"

corrected my incorrect mentions of the devfile version (2.1.0) to 2.0.0

MichalMaler committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
removed [NOTE] title

max-cx committed:
edited NOTE

max-cx committed:
Added 'For more information, see"

max-cx committed:
minor edit

max-cx committed:
minor edit

max-cx committed:
minor edit

max-cx committed:
added a missing 'a'

max-cx committed:
minor edit

max-cx committed:
minor edit

max-cx committed:
minor edits

max-cx committed:
minor edit

max-cx committed:
changed spec version from 2.1.0 to 2.0.0

MichalMaler committed:
Merge branch 'master' into 2999max-cx

MichalMaler committed:
Merge branch 'master' into 2999max-cx

MichalMaler commented on modules/end-user-guide/pages/creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -1,5 +1,5 @@
 [id="creating-a-workspace-by-importing-the-source-code-of-a-project"]
-:navtitle: Creating a workspace by importing the source code of a project
+:navtitle: Creating a workspace with your project's source code
```
I would keep `importing` in here. It is possible to rephrase it better, but I would definitively go with `importing` :)

MichalMaler commented on modules/end-user-guide/partials/assembly_workspaces-overview.adoc:
```diff
@@ -7,12 +7,12 @@
-. A web-based integrated development environment (IDE).
-. Tool dependencies, needed by developers to work on a project.
-. Application runtime: a replica of the environment where the application runs in production.
+. Your project's source code
```
These all need to be put in a bullet point list (using `*` instead of `.`) with the dot at the end.

max-cx replied:
:) Thanks! Periods made sense in the previous version, but asterisks make more sense in the updated version. Changed.
About the missing dots at the end, those are missing on purpose: IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation._"

MichalMaler commented on modules/end-user-guide/partials/proc_adding-language-support-plug-in-to-the-che-workspace.adoc:
```diff
@@ -5,51 +5,58 @@
++
+To install an instance of {prod-short}, see {link-installing-an-instance}.
 
+* A created workspace that is defined in this instance of {prod}
```
missing dot

max-cx replied:
:) On purpose. IBM Style Guide: p. 89 "_Write lists so that all or none of the items start with complete sentences. ... If list items comprise only sentence fragments, do not include any end punctuation_."

MichalMaler commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -107,35 +149,54 @@ image::workspaces/configure-command.png[Configure tasks]
 image::workspaces/{project-context}-welcome.png[Welcome screen]
 
-. Open the command palette by using `F1`, `CTRL-SHIFT-P`, or the link displayed on the Welcome screen.
+* To open the command palette, press F1 or Ctrl+Shift+P.
```
or kbd:[Ctrl+Shift+p].

max-cx replied:
Great, thank you, I didn't know about this! Changed.

max-cx commented on modules/end-user-guide/partials/proc_creating-a-workspace-by-importing-the-source-code-of-a-project.adoc:
```diff
@@ -3,49 +3,62 @@
 :parent-context-of-creating-a-workspace-by-importing-the-source-code-of-a-project: {context}
 
 [id="creating-a-workspace-by-importing-the-source-code-of-a-project_{context}"]
-= Creating a workspace by importing the source code of a project
```
This heading is incorrect because this file contains four sections of which three describe importing the source code into an already created workspace = incompatible with this heading. In the one section where the source code is imported at workspace creation, this heading is still incorrect because the act of importing the source code does not create the workspace. See my proposal on the next line below.

max-cx replied:
@rkratky, your suggestion is the best in my opinion:
**Importing the source code of a project into a workspace** covers all four sub-sections.
@MichalMaler, should I make this heading change then?

MichalMaler replied:
`Importing the source code of a project into a workspace` sounds like the heading of the week :) Go for it

max-cx replied:
Done.

max-cx committed:
fixed devfile version to 2.0.0

max-cx committed:
kbd:[Ctrl+Shift+P]

max-cx committed:
replaced .'s with *'s

max-cx committed:
devfile v2.1.0 > 2.0.0

max-cx committed:
replaced .'s with *'s

max-cx committed:
"workspace sample" > "code sample"

reverted back to the original: replaced "workspace sample" with original "code sample"

max-cx committed:
"workspace sample" > "code sample"

reverted back to the original: replaced "workspace sample" with the original "code sample"

max-cx committed:
replaced ...workspace-sample with ...a-code-sample

max-cx committed:
added (kbd:[Cmd+Shift+P] on macOS)

max-cx committed:
renamed file proc_creating-a-workspace-from-workspace-sample.adoc

max-cx committed:
renamed file proc_creating-a-workspace-from-workspace-sample.adoc

max-cx committed:
workspace sample > a code sample

max-cx committed:
workspace samples > code samples

max-cx commented on modules/end-user-guide/partials/assembly_creating-a-workspace-from-included-options.adoc:
```diff
@@ -0,0 +1,20 @@
+:parent-context-of-creating-a-workspace-from-included-options: {context}
+
+[id="creating-a-workspace-from-included-options_{context}"]
+= Creating a workspace from included options
```
This heading (and file name) used to be **Creating a workspace from code sample**.
There are two lower-level headings here, one of which I renamed **Creating a workspace from a code sample**.
The other lower-level heading I named **Creating a workspace from a template devfile**.
This means we cannot keep the top-level heading **Creating a workspace from code sample** because it does not reflect both lower-level headings and because in such case the same heading would be duplicated at two levels.

The following is my previously proposed, outdated solution (see my next comment):
```
As a solution, I am proposing a new top-level heading here:
Previously **Creating a workspace from a code sample**
becomes **Creating a workspace from included options**,
followed by the intro sentence:
_{prod-short} includes optional code samples and devfile templates to simplify a workspace creation._
This should meet the needs of the first-time user who needs to understand two basic user-centered ideas here:
1. Che in fact has such **included** (build-in) features as codes samples and devfile templates that will simplify the user's job of a workspace creation in a "production-grade" user story. (In contrast, "code sample" doesn't convey the idea of an important feature that can benefit the user in production.)
2. These included features are optional, hence **included options**.
Also, since the UI labels keep changing frequently, having this particular title rather generic can avoid more changes to it and bring about some permanency at this heading level (a benefit for regular readers).
```

max-cx replied:
@rkratky and @MichalMaler, I propose that we elevate the second heading (**Creating a workspace from a template devfile**) so that instead of one heading and two subheadings we get just two headings at the upper level.

MichalMaler replied:
Sounds like an idea we could all agree on. Nice. GO fo it :)

max-cx committed:
with > by using

max-cx committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
removed 'now'

removing 'now' to resolve an error on **Validate language on files added or modified**
modules/end-user-guide/partials/proc_adding-language-support-plug-in-to-the-che-workspace.adoc
59:29   error       Avoid using 'now'.              Vale.Avoid

MichalMaler committed:
Merge branch 'master' into 2999max-cx

MichalMaler committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
[2999] Heading renaming and substitution

Renamed and substituted the heading/file name
'Creating a workspace by importing the source code of a project'
with
the heading and file name
'Importing the source code of a project into a workspace'
throughout the repo.

Bug: https://issues.redhat.com/browse/RHDEVDOCS-2999

MichalMaler committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
[2999] Heading renaming and substitution continued

1. Renamed and elevated the heading 'Creating a workspace from Get Started view of user dashboard'.
2. Renamed and elevated the heading 'Creating a custom workspace from a template'.
3. Removed the now redundant-level heading 'Creating a workspace from code sample'.
4. Adjusted modular docs throughout the repo to account for these changes.

Bug: https://issues.redhat.com/browse/RHDEVDOCS-2999

MichalMaler committed:
Merge branch 'master' into 2999max-cx

MichalMaler committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
removed one blank line

max-cx committed:
removed a blank line

MichalMaler committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
[2999] Fix: some xrefs

Removed text from inside of some square brackets in the files that have been edited for this PR.

Bug: https://issues.redhat.com/browse/RHDEVDOCS-2999

themr0c commented on modules/end-user-guide/pages/creating-a-workspace-from-a-code-sample.adoc:
```diff
@@ -0,0 +1,7 @@
+
+:navtitle: Creating a workspace from a code sample
+:keywords: end-user-guide, creating-a-workspace-from-a-code-sample
+:page-aliases: .:creating-a-workspace-from-a-code-sample
```
```suggestion
:page-aliases: .:creating-a-workspace-from-a-code-sample, .:creating-a-workspace-from-code-sample, creating-a-workspace-from-code-sample
```

themr0c replied:
When renaming pages, add page-aliases to handle redirections from the older page name.

themr0c commented on modules/end-user-guide/pages/importing-the-source-code-of-a-project-into-a-workspace.adoc:
```diff
@@ -0,0 +1,6 @@
+[id="importing-the-source-code-of-a-project-into-a-workspace"]
+:navtitle: Importing the source code of a project into a workspace
+:keywords: end-user-guide, importing-the-source-code-of-a-project-into-a-workspace
+:page-aliases: .:importing-the-source-code-of-a-project-into-a-workspace, importing-the-source-code-of-a-project-into-a-workspace
```
```suggestion
:page-aliases: .:importing-the-source-code-of-a-project-into-a-workspace, importing-the-source-code-of-a-project-into-a-workspace, .:creating-a-workspace-by-importing-the-source-code-of-a-project, creating-a-workspace-by-importing-the-source-code-of-a-project
```

themr0c replied:
When renaming pages, add page-aliases to handle redirections from the older page name.

themr0c commented on modules/end-user-guide/partials/assembly_authoring-devfiles-version-2.adoc:
```diff
@@ -1,21 +1,34 @@
 
 [id="authoring-devfiles-version-2_{context}"]
-= Authoring devfiles version 2
+= Authoring a devfile 2.0.0
```
I believe Che always follows the latest version of the devfile 2 specifications, so putting the minor and patch version here is not a feature, we should keep reference to "devfile 2". @l0rd can you confirm?

max-cx committed:
Update modules/end-user-guide/pages/importing-the-source-code-of-a-project-into-a-workspace.adoc

Co-authored-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

max-cx committed:
Merge branch 'master' into 2999max-cx

themr0c commented on modules/end-user-guide/pages/importing-the-source-code-of-a-project-into-a-workspace.adoc:
```diff
@@ -0,0 +1,6 @@
+[id="importing-the-source-code-of-a-project-into-a-workspace"]
+:navtitle: Importing the source code of a project into a workspace
+:keywords: end-user-guide, importing-the-source-code-of-a-project-into-a-workspace
+:page-aliases: .:importing-the-source-code-of-a-project-into-a-workspace, importing-the-source-code-of-a-project-into-a-workspace, .:creating-a-workspace-by-importing-the-source-code-of-a-project, creating-a-workspace-by-importing-the-source-code-of-a-project
```
```suggestion
:page-aliases: .:importing-the-source-code-of-a-project-into-a-workspace, .:creating-a-workspace-by-importing-the-source-code-of-a-project, creating-a-workspace-by-importing-the-source-code-of-a-project
```

themr0c committed:
Update modules/end-user-guide/pages/importing-the-source-code-of-a-project-into-a-workspace.adoc

themr0c committed:
Update modules/end-user-guide/pages/creating-a-workspace-from-a-code-sample.adoc

max-cx committed:
[2999] fix: correct '.adoc.adoc'

in:
modules/end-user-guide/partials/assembly_workspaces-overview.adoc:73:* xref:creating-a-workspace-from-a-code-sample.adoc.adoc[]

Bug: https://issues.redhat.com/browse/RHDEVDOCS-2999

max-cx committed:
minor edits

max-cx committed:
fix: update the UI label

The UI label was changed after this PR was opened.

max-cx committed:
fix: correct the procedure

Correcting the procedure due to the default behavior in the UI.

max-cx committed:
fix: align reference to other sections of docs

max-cx committed:
fix: add .Example syntax

max-cx committed:
fix: replace 'section' with 'component'

max-cx committed:
fix: align a reference with other sections of docs

max-cx committed:
.

max-cx committed:
fix: 'section' > 'component'

max-cx committed:
fix: adjusted procedure to account for default behavior of UI

max-cx committed:
fix: example line

max-cx committed:
fix: align reference format with other sections of docs

max-cx committed:
fix: eliminating fluff

max-cx committed:
fix: example text

max-cx committed:
fix: 'component

max-cx committed:
fix: 'component' > 'section'

Although 'devfile component' is mentioned in the docs, https://docs.devfile.io/devfile/2.1.0/user-guide/index.html mentions 'devfile sections'.

max-cx committed:
fix: rewriting example text

max-cx committed:
fix: minor text improvements

max-cx committed:
fix: moved some example text into procedure

max-cx committed:
fix: fix broken ol

max-cx committed:
fix: improve layout of .Prerequisites

max-cx committed:
fix: delete whitespaces

max-cx committed:
fix: delete a whitespace

max-cx committed:
fix: fix ol

max-cx committed:
Merge branch 'master' into 2999max-cx

max-cx committed:
fix: remove trailing periods on `.Example` lines

max-cx committed:
fix: remove trailing period on `.Example` line

max-cx committed:
Merge branch 'master' into 2999max-cx
